node/overseer/overseer-gen/proc-macro/src/impl_builder.rs

```diff
@@ -210,56 +203,53 @@ pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 			}
 		});
 	// Produce setters for all baggage fields as well
-	let baggage_specific_setters = info.baggage()
-		.iter()
-		.enumerate()
-		.map(|(idx, bag_field)| {
-			// Baggage fields follow subsystems
-			let baggage_idx = idx + subsystem_name.len();
-			let fname = &bag_field.field_name;
-			let ftype = &bag_field.field_ty;
-			let impl_generics = placeholder_type[..baggage_idx]
-				.iter()
-				.chain(placeholder_type[baggage_idx+1..].iter());
-			let other_field_name = field_name[..baggage_idx]
-				.iter()
-				.chain(field_name[baggage_idx+1..].iter());
-
-			let uninit_generics = produce_setter_generic_permutation(
-				ftype,
-				baggage_idx,
-				placeholder_type.as_slice(),
-				&Ident::new("OverseerFieldUninit", Span::call_site()));
-			let return_type_generics = produce_setter_generic_permutation(
-				ftype,
-				baggage_idx,
-				placeholder_type.as_slice(),
-				&Ident::new("OverseerFieldInit", Span::call_site()));
-			let additional_baggage_generic = if bag_field.generic {
-				quote!{#ftype,}
-			}
-			else {
-				quote!()
-			};
-
-			quote! {
-				impl <S, #additional_baggage_generic #( #impl_generics, )*>
-				#builder <S, #( #uninit_generics, )*> #builder_where_clause {
-					/// Specify the baggage in the builder
-					pub fn #fname (self, var: #ftype ) ->
-						#builder <S, #( #return_type_generics, )*>
-					{
-						#builder {
-							#fname: OverseerFieldInit::<#ftype>::Value(var),
-							#(
-								#other_field_name: self.#other_field_name,
-							)*
-							spawner: self.spawner,
-						}
+	let baggage_specific_setters = info.baggage().iter().enumerate().map(|(idx, bag_field)| {
+		// Baggage fields follow subsystems
+		let baggage_idx = idx + subsystem_name.len();
+		let fname = &bag_field.field_name;
+		let ftype = &bag_field.field_ty;
+		let impl_generics = placeholder_type[..baggage_idx]
+			.iter()
+			.chain(placeholder_type[baggage_idx + 1..].iter());
+		let other_field_name =
+			field_name[..baggage_idx].iter().chain(field_name[baggage_idx + 1..].iter());
```

Comment:
```suggestion
			field_name.iter().enumerate().filter(|(idx,_)| { baggage_idx != idx });
```

Reply:
Well, it would be `field_name.iter().enumerate().filter(|(idx,_)| { baggage_idx != *idx }).map(|(_, field)| field)` which looks much cryptic to me. Alternative is `let mut other_field_name = field_name.clone()` followed by `other_field_name.remove(idx)` which is simple but not very smart in general.

Reply:
Good point, looking `derive_replacable_generic_lists` this must be easier to achieve, I find this code rather convoluted and there must be an easier to comprehend structure.